Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -1445,87 +1445,184 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::sort)
 
+            // ii. Let to be ! ToString(𝔽(k + itemCount)).
+            auto to = i + insert_count;
+
+            // iii. If ? HasProperty(O, from) is true, then
             if (TRY(this_object->has_property(from))) {
+                // 1. Let fromValue be ? Get(O, from).
                 auto from_value = TRY(this_object->get(from));
+
+                // 2. Perform ? Set(O, to, fromValue, true).
+                // FIXME: O != to
                 TRY(this_object->set(to, from_value, Object::ShouldThrowExceptions::Yes));
-            } else {
+            }
+            // iv. Else,
+            else {
+                // 1. Perform ? DeletePropertyOrThrow(O, to).
+                // FIXME: O != to
                 TRY(this_object->delete_property_or_throw(to));
             }
+
+            // v. Set k to k + 1.
         }
 
-        for (u64 i = initial_length; i > new_length; --i)
+        // c. Set k to len.
+        // d. Repeat, while k > (len - actualDeleteCount + itemCount),
+        // FIXME: loop index should be k
+        // FIXME loop end not spec logic
+        for (u64 i = initial_length; i > new_length; --i) {
+            // i. Perform ? DeletePropertyOrThrow(O, ! ToString(𝔽(k - 1))).
             TRY(this_object->delete_property_or_throw(i - 1));
-    } else if (insert_count > actual_delete_count) {
+
+            // ii. Set k to k - 1.
+        }
+    }
+    // 17. Else if itemCount > actualDeleteCount, then
+    else if (insert_count > actual_delete_count) {
+        // a. Set k to (len - actualDeleteCount).
+        // b. Repeat, while k > actualStart,
+        // FIXME: loop index should be k
         for (u64 i = initial_length - actual_delete_count; i > actual_start; --i) {
+            // FIXME: not property key
+            // FIXME: random index name
+
+            // i. Let from be ! ToString(𝔽(k + actualDeleteCount - 1)).
             u64 from_index = i + actual_delete_count - 1;
+
+            // ii. Let to be ! ToString(𝔽(k + itemCount - 1)).
             auto to = i + insert_count - 1;
 
+            // iii. If ? HasProperty(O, from) is true, then
             if (TRY(this_object->has_property(from_index))) {
+                // 1. Let fromValue be ? Get(O, from).
                 auto from_value = TRY(this_object->get(from_index));
+
+                // 2. Perform ? Set(O, to, fromValue, true).
                 TRY(this_object->set(to, from_value, Object::ShouldThrowExceptions::Yes));
-            } else {
+            }
+            // iv. Else,
+            else {
+                // 1. Perform ? DeletePropertyOrThrow(O, to).
                 TRY(this_object->delete_property_or_throw(to));
             }
+
+            // v. Set k to k - 1.
         }
     }
 
-    for (u64 i = 0; i < insert_count; ++i)
+    // 18. Set k to actualStart.
+    // 19. For each element E of items, do
+    // FIXME loop index should start at actual_start
+    // FIXME loop index should be k
```

Comment:
we can't range-for loop over `items` here so I don't think these FIXMEs make sense